### PR TITLE
Redesign: New icons

### LINKED ISSUE #6299: Redesign: Update icons in the design system
## Task Description

- [x] Move design system 24x24 dashboard icons to dashboard icon folder
- [x] Remove all other other icons from design system
- [x] Export all new icons from Figma in 32x32 box
- [x] Import all new icons to design system
- [x] Update buttons to not scale icons, but keep 32x32 box centered regardless of button size

### PR DESCRIPTION
**Pull request by barklund**
## Context

See #6299 for todo.

### QA

- [x] This is a non-user-facing change and requires no QA

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6299

### CODE REVIEW

**Comment by barklund** [comment]
@samwhale, @BrittanyIRL, I've done the first part of this task - moved dashboard specific icons (that don't conform to the 32x32 icon size) out of the design system and into the dashboard. Please review, thanks!

**Review by BrittanyIRL — DISMISSED** [review]
Thanks for doing this, makes sense to me - especially after talking this morning about focusing on making the design system the shared parts of the project.

**Comment by barklund** [comment]
I have now replaced icons throughout the editor, and I believe things still look good.

I have implemented the new icons naming them for what they are rather than what they're supposed to be used for. I believe this follows regular icon library naming conventions.

### Please verify
All, please check:
- [ ] all icons in the design system in Storybook - they're also in color now!

@miina: please check:
- [x] all icons in the preset panels
- [ ] plus icon in the color picker preset actions.

@BrittanyIRL: please check:
- [x] chevron icons in the dropdown in the design system
- [x] close icon in the snack-bar in the design system
- [x] alert icon for the pre-publish checklist tab button (I also added an icon to the storybook for tabview)

@samwhale: please check
- [x] all icons in the help center, where I also updated the design for the toggle button somewhat

### Known bugs
Some icons look weird:

- page layouts icon in the element library
- notification icon in the publish button

These will be fixed when those components will be re-done anyway.

File: assets/src/design-system/icons/stories/index.js
Comment (BrittanyIRL) [inline assets/src/design-system/icons/stories/index.js:75]: 💅

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (BrittanyIRL) [inline assets/src/edit-story/components/helpCenter/toggle/index.js]: This feels odd? Could this not be added to the label as padding?

**Review by BrittanyIRL — APPROVED** [review]
Thanks for updating all of these at once. 

To me there's a little bit of duplicate here with direction differences, which if that's what we want to be the standard I am open to but I want to mention it. It would be simple enough to have `Arrow` and eliminate ArrowDown, ArrowRight, ArrowUp - same thing for ChevronUp vs Down, since that will probably almost always change based on a prop it wouldn't be that big of a deal to rely on css for the direction update. WDYT? 

Design system, dashboard, help and prepublish updates all checkout. I found 4 things that needed to be updated, made a PR for the updates to this branch to save you the busy work: https://github.com/google/web-stories-wp/pull/6319 if you agree with them. 

Not worried about the dropdown chevron because I have an update for that with search anyway since those styles changed. 

Approving with comments since we're in different timezones and I'd rather not slow this down.

File: docs/design-system.md
Comment (barklund) [inline docs/design-system.md:76]: @BrittanyIRL, can you approve of this change?

Comment (BrittanyIRL) [reply docs/design-system.md:76]: Thanks for adding documentation! Like the note about trying to maintain alphabetization, think this makes having the icons flat in one directory more manageable.

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (barklund) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: I like this approach - it also makes RTL simpler, as the divs just go in the other order instead of having to switch margin or padding from right to left. We've used it in a few places too. But I'm okay with changing it.

**Comment by barklund** [comment]
> To me there's a little bit of duplicate here with direction differences, which if that's what we want to be the standard I am open to but I want to mention it. It would be simple enough to have `Arrow` and eliminate ArrowDown, ArrowRight, ArrowUp - same thing for ChevronUp vs Down, since that will probably almost always change based on a prop it wouldn't be that big of a deal to rely on css for the direction update. WDYT?

I just don't want to assume anything of the icons - they might be slightly different in different directions and they're allowed to be. Of course, if it's a rotating icon, we have to use the same one, but if not, I prefer having separate icons. The SVG's are so few bytes each, that it's really not a performance worry, IMO.

I'll wait for @miina's input in the morning and then merge it tomorrow once approved by her as well - I did make substantial edits [in the `<EmptyPanel />`](https://github.com/google/web-stories-wp/pull/6311/files#diff-6d3d2a7fc02a3d05f1d56d1c134102f2d678f67f4ed33cacc614080d1aea9aa6R33-R37) in particular, so just want to check I didn't mess something up there.

I'll for now merge this into another PR (#6204) to continue work on that though, but that's for redesign so it'll hold until 1.4 is out the door anyway.

**Comment by miina** [comment]
@barklund Two findings:

1. This doesn't look right-sized:
Screenshot 2021-02-11 at 11 40 57

2. The + icon inside the text [should be smaller](https://xwp.slack.com/archives/C01ESBB8ZN1/p1612366572007000?thread_ts=1612345243.006500&cid=C01ESBB8ZN1):
Screenshot 2021-02-11 at 11 42 01

Should be (LMK if that has changed though):
Screen Shot 2021-02-03 at 7 34 19 AM

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (BrittanyIRL) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: ah, if RTL is the reason then I'm ok with it. I was unaware that this was a standard in the editor.

**Comment by BrittanyIRL** [comment]
> I just don't want to assume anything of the icons - they might be slightly different in different directions and they're allowed to be. Of course, if it's a rotating icon, we have to use the same one, but if not, I prefer having separate icons. The SVG's are so few bytes each, that it's really not a performance worry, IMO.

This is a good reason in my book! 👍

File: docs/design-system.md
Comment (samwhale) [inline docs/design-system.md:82]: Should we add a step here to compress the svg first?

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (samwhale) [inline assets/src/edit-story/components/helpCenter/toggle/index.js]: So now the chevron only shows after the notifications are gone. Is that intended?
image

Comment (samwhale) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: image

**Review by samwhale — APPROVED** [review]
Icons in storybook look good. Help center looks good - but wondering if the chevron should still be shown when there are unread notifications

File: assets/src/design-system/components/toggle/index.js
Comment (samwhale) [inline assets/src/design-system/components/toggle/index.js]: 💭 do you like the checkmark in the current position or up 1 pixel?

checkmark-position

Comment (samwhale) [reply assets/src/design-system/components/toggle/index.js]: To me it feels like it's better centered when it's slightly higher (maybe even 0.5 px), but I'll leave that to your judgement

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (BrittanyIRL) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: it is, designs were updated very recently! https://xwp.slack.com/archives/C01ESBB8ZN1/p1612994877142300

File: docs/design-system.md
Comment (BrittanyIRL) [reply docs/design-system.md:82]: optimization is happening by default in webpack - https://github.com/google/web-stories-wp/pull/6287#discussion_r573089953

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (barklund) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: Yes, the designs have been updated to [this](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=5663%3A3546). Isn't it lovely with a moving target? 😁 

But I know see, that it has actually updated again since this implementation for the narrow viewport. Here "Help Center" is replaced by just the question mark icon followed by either the unread count or the chevron.

File: docs/design-system.md
Comment (barklund) [reply docs/design-system.md:82]: That supposed to happen automagically in the build pipeline, but I'm actually not sure if we still do this.

@swissspidy, can you confirm that this happens? I only see `svgr` in the [Webpack config](https://github.com/google/web-stories-wp/blob/main/webpack.config.cjs#L95), but no mention of `svgo`.

[This doc](https://react-svgr.com/docs/options/#svgo) indicates that `svgo` is standard when using `svgr`, so I assume it works?

File: assets/src/design-system/components/toggle/index.js
Comment (barklund) [reply assets/src/design-system/components/toggle/index.js]: Great catch! I've forwarded the question to UX.

**Comment by barklund** [comment]
> 1. This doesn't look right-sized:
>    Screenshot 2021-02-11 at 11 40 57

As [mentioned](https://github.com/google/web-stories-wp/pull/6311#issuecomment-777086259). But now that I think about, that's not good enough. This PR goes into 1.4. The library header fix does not. I'll fix in this PR. Thanks for reminding me.

> 2. The + icon inside the text [should be smaller](https://xwp.slack.com/archives/C01ESBB8ZN1/p1612366572007000?thread_ts=1612345243.006500&cid=C01ESBB8ZN1):
>    Screenshot 2021-02-11 at 11 42 01
> 
> Should be (LMK if that has changed though):
> Screen Shot 2021-02-03 at 7 34 19 AM

I'll fix it. Again my thinking was, that we're redoing the color picker anyway, but not for this release, so yes, this should be fixed now.

File: assets/src/design-system/components/toggle/index.js
Comment (samitron7) [reply assets/src/design-system/components/toggle/index.js]: Can we move it one pixel up. looks more centered.

File: docs/design-system.md
Comment (swissspidy) [reply docs/design-system.md:82]: It should still be the default, yes. There's also a `.svgo.yml` config file for it that - at least when I last tested it - was applied to SVGs.

You could quickly test this yourself by removing those configs and re-building the app. It should cause the icons in the app to appear rather broken :-).

If it doesn't, then there's a bug.

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (BrittanyIRL) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: oh heck, i thought it was that on narrow viewport it's just 'help' + the chevron or badge depending on if there's unread tips? Either way, this update's not going in until after the release is cut so if it's off a little bit when this is merged it's ok because we can just have a folllow up ticket/pr to actually address that. I feel like adjusting this FTUE here is just scope creep.

Comment (barklund) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: This will actually go in, as this mainly updates icons in the design system, and only very few icons in the actual editor - and none of those are changed visually, they're merely updated in sizing and bounding box.

File: assets/src/design-system/components/toggle/index.js
Comment (barklund) [reply assets/src/design-system/components/toggle/index.js]: Fixed in e08c625c6.

File: assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js
Comment (barklund) [inline assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js:226]: @miina, while looking to fix this, I noticed that the actual **+** used in the design is a real text plus just in a bigger font and in white. So I did the same here and it looks pretty similar:

| Design | Implementation |
|-|-|
| Screen Shot 2021-02-11 at 20 49 07 | Screen Shot 2021-02-11 at 20 48 57 |

File: assets/src/edit-story/components/helpCenter/toggle/index.js
Comment (barklund) [reply assets/src/edit-story/components/helpCenter/toggle/index.js]: Implemented the new designs in 629038057. Feel free to take them for a spin - I think it looks pretty good. Also, see follow-up discussion [here](https://xwp.slack.com/archives/C01ESBB8ZN1/p1613098933195200).

**Comment by barklund** [comment]
> > 1. This [page layout icon] doesn't look right-sized:
>
> I'll fix in this PR.
> 
> > 1. The + icon inside the text should be smaller
> 
> I'll fix it.

And done.

File: assets/src/edit-story/components/library/panes/pageLayouts/pageLayoutsIcon.js
Comment (barklund) [inline assets/src/edit-story/components/library/panes/pageLayouts/pageLayoutsIcon.js:29]: This is a purely temporary (and quite magic - `sqrt(2)` even!) value until #6207 is implemented. This will go out for 1.4, but should be fine.

File: assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js
Comment (miina) [reply assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js:226]: Sounds good to me, the approach. Looks like it needs a bit more room though and looks thicker than on the design. Maybe reduce the font size to 20 and/or make it inline-block to assign fixed width? (I can look at it within #6208 as well though, not critical!)

**Review by miina — APPROVED** [review]

File: assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js
Comment (barklund) [reply assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js:226]: Let's deploy as-is for 1.4 and we can fix in #6208 when redoing the component anyway.

**Pull request merged**
